server/src/main/scala/org/http4s/server/SSLClientAuthMode.scala

```diff
@@ -0,0 +1,8 @@
+package org.http4s.server
+
+/**
+  * Client Auth mode for mTLS
+  */
+object SSLClientAuthMode extends Enumeration {
```

Comment:
Is there some Java interop reason to use an `Enumeration` instead of a sealed trait here?  I imagine:

```
sealed abstract class SSLClientAuthMode extends Product with Serializable
object SSLClientAuthMode {
  case object NotRequested extends SSLClientAuthMode
  case object Requested extends SSLClientAuthMode
  case object Required extends SSLClientAuthMode
}
```

I vaguely remember something about Scala enums and Java interop, but not whether it's good or bad.  But I don't think this is going to be called from Java anyway.

Reply:
I generally used `sealed trait` but otherwise the same

Reply:
Will convert to a trait